apps/full-site-editing/full-site-editing-plugin/full-site-editing/utils/replace-template-parts.php

```diff
@@ -58,11 +65,27 @@ function a8c_fse_replace_template_parts( $html ) {
 
 		$node_to_replace = $candidate_nodes[0];
 
-		@$temp_doc->loadHTML( mb_convert_encoding( $replacement['fse_content'], 'HTML-ENTITIES', 'UTF-8' ) );
+		$encoded_fse_content = mb_convert_encoding( $replacement['fse_content'], 'HTML-ENTITIES', 'UTF-8' );
+
+		// We wrap the FSE html content in a div, since LibXML requires a root node.
+		$wrapped_fse_content = '<div>' . $encoded_fse_content . '</div>';
+
+		// LIBXML_HTML_NOIMPLIED keeps the HTML intact, without adding html, body or other unnecessary tags in a single node.
+		@$temp_doc->loadHTML( $wrapped_fse_content, LIBXML_HTML_NOIMPLIED );
 
 		$fse_node = $doc->importNode( $temp_doc->documentElement, true );
-		$node_to_replace->parentNode->replaceChild( $fse_node, $node_to_replace );
+
+		// Remove all the children of the element to be replaced in order to keep using the same container for the FSE content.
+		while ( $node_to_replace->hasChildNodes() ) {
+			$node_to_replace->removeChild( $node_to_replace->firstChild );
+		}
+
+		// Rather than appending the FSE content node, we add its child nodes so the extra wrapping div added above is not rendered.
+		for ( $i = 0; $i < $fse_node->childNodes->length; $i++ ) {
```

Comment:
Seems this is the problematic code causing the weird behavior with the footer template parts I noted on https://github.com/Automattic/wp-calypso/pull/34171#issuecomment-504298538

Reverting this back to `$node_to_replace->parentNode->replaceChild( $fse_node, $node_to_replace );` solves the problem, but we have an extra `div` wrapping the template part content and we lose the `.site-header`/`.site-footer` elements.

Maybe that's not a big deal, so anyone is welcome to change it.

Reply:
Fixed this for now in 2503f698c1.